### DISCUSSION TITLE
Need help for setting up Honeypots and Kibana in kubernetes/docker

### DISCUSSION THREAD

**Discussion by devamtech** [telekom-security/tpotce#821] [Q&A]
HI 

I am trying to set up honey pots in docker and getting errors. Can you please provide any documentation or Please provide Kubernetes yaml file

Many thanks in advance.

Regards
Deva

**Comment by Ganesh-403** [comment]
@devamtech 
At the moment, **T-Pot does not provide official Kubernetes manifests or Helm charts**.

The project is designed around **Docker and Docker Compose**, and the installation, customization, and lifecycle management all assume a Docker Compose deployment. The official documentation only provides Compose-based installation and customization workflows. :contentReference[oaicite:0]{index=0}

If you're simply looking to run T-Pot with Docker, I'd recommend following the documented approach:

```bash
git clone https://github.com/telekom-security/tpotce
cd tpotce
./install.sh
```

or, for platforms like macOS/Windows (Docker Desktop), use one of the provided compose files:

```bash
cp compose/mac_win.yml ./docker-compose.yml
docker compose up -d
```

T-Pot also includes several predefined Compose configurations (`standard.yml`, `mini.yml`, `sensor.yml`, etc.) and a `customizer.py` script if you only want a subset of honeypots. :contentReference[oaicite:1]{index=1}

### If your goal is Kubernetes

From what I can tell, there is **no official Kubernetes deployment** for T-Pot.

While it is technically possible to containerize individual honeypots and deploy them to Kubernetes, T-Pot relies on several Docker-specific behaviors, including:

- Docker Compose networking
- host port bindings
- persistent volumes
- container orchestration between Logstash, Elasticsearch, Kibana, Nginx, and the honeypots

Because of that, there isn't a supported Kubernetes YAML that you can apply directly. Converting the Compose stack with a tool such as `kompose` may help as a starting point, but you'll likely need to manually adapt networking, storage, and privileged/container settings afterward. :contentReference[oaicite:2]{index=2}

Could you clarify what you're trying to deploy?

- **The complete T-Pot platform** (ELK + Kibana + all honeypots)?
- **Only one or two honeypots** (e.g. Cowrie, Dionaea) in Kubernetes?
- Or are you looking to deploy **Kibana/Elasticsearch separately** and run the honeypots elsewhere?

That will determine the most practical deployment approach.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
